azurelinuxagent/ga/exthandlers.py

```diff
@@ -783,6 +783,19 @@ def __init__(self, ext_handler, protocol):
             self.logger.error(u"Failed to create extension log dir: {0}", e)
 
         log_file = os.path.join(self.get_log_dir(), "CommandExecution.log")
+
+        try:
+            if os.stat(log_file).st_size > execution_log_max_size:
+                with open(log_file, "rb") as existing_log_file_buffer:
+                    existing_log_file_buffer.seek(-1 * execution_log_max_size, 2)
+                    _ = existing_log_file_buffer.readline()
```

Comment:
Out of curiosity, why is this needed?

Reply:
Are you referring specifically to L#791 itself?
If so,  it's just an easy way to get the log file at a size less than the max, but on a line ending.

Reply:
Ahh very cool, interesting hack haha